packages/react/src/components/ListBuilder/ListBuilder.jsx

```diff
@@ -0,0 +1,219 @@
+  },
+};
+
+const ListBuilder = ({ testID, items, itemCount, selectedItems, i18n, onAdd, onRemove }) => {
+  const mergedI18n = {
+    ...defaultProps.i18n,
+    ...i18n,
+  };
+  const { allListTitle, selectedListTitle } = mergedI18n;
+
+  const handleAdd = useCallback(
+    (id) => (event) => {
+      event.persist();
+      onAdd(event, id);
+    },
+    [onAdd]
+  );
+  const handleRemove = useCallback(
+    (id) => (event) => {
+      event.persist();
+      onRemove(event, id);
+    },
+    [onRemove]
+  );
+
+  const allListItems = useMemo(
+    () =>
+      items?.map((item) => {
+        const rowActions =
+          item.content.rowActions !== undefined
+            ? item.content.rowActions
+            : () => [
+                <Button
+                  key={`${item.id}-list-item-button`}
+                  testID={`${testID}-add-button-${item.id}`}
+                  role="button"
+                  aria-label={mergedI18n.addLabel}
+                  renderIcon={ArrowRight16}
+                  hasIconOnly
+                  kind="ghost"
+                  size="small"
+                  onClick={handleAdd(item.id)}
+                  iconDescription={mergedI18n.addLabel}
+                />,
+              ];
+
+        return {
+          ...item,
+          content: {
+            ...item.content,
+            rowActions,
+          },
+        };
+      }) ?? [],
+    [handleAdd, items, mergedI18n.addLabel, testID]
+  );
+
+  const selectedListItems = useMemo(
+    () =>
+      selectedItems?.map((selectedItem) => {
+        const rowActions =
+          selectedItem.content.rowActions !== undefined
+            ? selectedItem.content.rowActions
+            : () => [
+                <Button
+                  key={`${selectedItem.id}-list-item-button`}
+                  testID={`${testID}-add-button-${selectedItem.id}`}
+                  role="button"
+                  aria-label={mergedI18n.removeLabel}
+                  renderIcon={ArrowLeft16}
+                  hasIconOnly
+                  kind="ghost"
+                  size="small"
+                  onClick={handleRemove(selectedItem.id)}
+                  iconDescription={mergedI18n.removeLabel}
+                />,
+              ];
+
+        return {
+          ...selectedItem,
+          content: {
+            ...selectedItem.content,
+            rowActions,
+          },
+        };
+      }) ?? [],
+    [handleRemove, mergedI18n.removeLabel, selectedItems, testID]
+  );
+
+  return (
+    <div className={`${iotPrefix}--list-builder__container`}>
+      <div className={`${iotPrefix}--list-builder__all`} data-testid={`${testID}__all`}>
+        <HierarchyList
+          title={allListTitle(itemCount ?? allListItems.length)}
+          items={allListItems}
+          hasSearch
+          hasPagination={false}
+          searchId={`${iotPrefix}--list-builder__all--search`}
+          i18n={{
```

Comment:
How do we change the other i18n strings in the HierarchyList if we for instance want to support another language?

Reply:
Good catch. I'll add expand/close. The others aren't used b/c they aren't exposed.